linera-chain/src/chain.rs

```diff
@@ -208,20 +208,26 @@ where
     /// Verifies that this chain is up-to-date and all the messages executed ahead of time
     /// have been properly received by now.
     pub async fn validate_incoming_messages(&mut self) -> Result<(), ChainError> {
+        use futures::stream::{self, StreamExt, TryStreamExt};
         let chain_id = self.chain_id();
         let origins = self.inboxes.indices().await?;
         let inboxes = self.inboxes.try_load_entries(&origins).await?;
-        for (origin, inbox) in origins.into_iter().zip(inboxes) {
-            let event = inbox.removed_events.front().await?;
-            ensure!(
-                event.is_none(),
-                ChainError::MissingCrossChainUpdate {
-                    chain_id,
-                    origin: origin.into(),
-                    height: event.unwrap().height,
-                }
-            );
-        }
+        let stream = origins.into_iter().zip(inboxes);
+        let stream = stream::iter(stream)
```

Comment:
It would be great (in the long term, with pagination and stuff) to have a `stream` method to replace these four lines: `let stream = self.inboxes.stream()`
That would also not actually load anything yet, so that the `map` can be applied to that stream directly, and e.g. one entry can be checked for events while the other entry's inbox is still being loaded.

Reply:
Ok, will create a task for this. We were thinking about it when the load_entries was implemented, but it was judged too complicated at the time.